src/core/sequence_batch_scheduler.cc

```diff
@@ -1197,11 +1197,25 @@ OldestSequenceBatch::OldestSequenceBatch(
     preferred_batch_sizes.insert(size);
   }
 
+  // Need to enforce equal shape batches (i.e. non-ragged batches)
+  // if the model allows one or more variable-size input tensors or
+  // has shape-tensor inputs. This is not needed if all input shapes
+  // are non-variable and if there are no shape tensors... so we
+  // don't enable it in that case for efficiency reasons.
+  std::unordered_map<std::string, bool> enforce_equal_shape_tensors;
+  for (const auto input : config.input()) {
+    if (input.is_shape_tensor()) {
+      enforce_equal_shape_tensors.insert({input.name(), true});
+    } else if (GetElementCount(input) == -1) {
+      enforce_equal_shape_tensors.insert({input.name(), false});
```

Comment:
Put a comment here that this should be "true" but some existing custom backend implementations rely on ragged batches being allowed so need "false" until DLIS-904 is fixed.